core/valor_core/schemas.py

```diff
@@ -0,0 +1,1739 @@
+            )
+
+        if len(coordinates) > 0 and not (
+            isinstance(coordinates[0], list)
+            and len(coordinates[0]) > 0
+            and isinstance(coordinates[0][0], dict)
+            and all(
+                all(set(pt.keys()) == {"x", "y"} for pt in contour)
+                for contour in coordinates
+            )
+        ):
+            raise TypeError(
+                "Coordinates should either be an empty list, or it should be a list of lists containing dictionaries with 'x' and 'y' keys."
+            )
+
+        if not (
+            all(
+                all(pt["x"] >= 0 and pt["y"] >= 0 for pt in contour)
+                for contour in coordinates
+            )
+        ):
+            raise ValueError(
+                "Coordinates cannot be negative when converting to a raster."
+            )
+
+        contours = [
+            [(min(pt["x"], width), min(pt["y"], height)) for pt in contour]
+            for contour in coordinates
+        ]
+
+        img = Image.new("1", (width, height), 0)
+
+        for contour in contours:
+            if len(contour) >= 2:
+                ImageDraw.Draw(img).polygon(contour, outline=1, fill=1)
+
+        return cls(np.array(img))
+
+    @classmethod
+    def from_geometry(cls, geometry: Box | Polygon, height: int, width: int):
+        """
+        Create a Raster object from a geometry.
+
+        Parameters
+        ----------
+        coordinates : list[list[dict[str, int]]]
+            Defines the bitmask as a nested list of coordinates.
+        height : int
+            The intended height of the binary mask.
+        width : int
+            The intended width of the binary mask.
+
+        Returns
+        -------
+        schemas.Raster
+        """
+        if not (isinstance(geometry, Box) or isinstance(geometry, Polygon)):
+            raise TypeError("Geometry should be a Box or Polygon.")
+
+        return cls.from_coordinates(
+            geometry.to_coordinates(), height=height, width=width
+        )
+
+
+@dataclass
+class Embedding:
+    """
+    Represents a model embedding.
+
+    Parameters
+    ----------
+    value : List[float], optional
+        An embedding value.
+    """
+
+    value: Optional[Union[list[int], list[float]]] = None
+
+    def __post_init__(self):
+        """Validate instantiated class."""
+
+        if not isinstance(self.value, list):
+            raise TypeError(
+                f"Expected type 'Optional[List[float]]' received type '{type(self.value)}'"
+            )
+        elif len(self.value) < 1:
+            raise ValueError("embedding should have at least one dimension")
+
+
+@dataclass
+class Datum:
+    """
+    A class used to store information about a datum for either a 'GroundTruth' or a 'Prediction'.
+
+    Attributes
+    ----------
+    uid : String
+        The UID of the datum.
+    metadata : Dictionary
```

Comment:
I think you are using the symbolic types for the docstrings in this file. Could you update with proper typing?